cmd/deploy/deploy.go

```diff
@@ -95,6 +96,7 @@ type DeployCommand struct {
 	PipelineType       model.Archetype
 	isRemote           bool
 	runningInInstaller bool
+	repositorySHA      string
```

Comment:
Define properties in a struct which won't be initialized as part of the instance creation is misleading. It might lead to panics (nil pointers. In this case it doesn't matter as it's not a pointer) or a wrong logic based on checks when the property wasn't set yet. I would try to use this pattern to pass the sha to the command configuration